sdk/storage/azblob/container/models.go

```diff
@@ -329,3 +332,79 @@ func formatTime(c *SignedIdentifier) error {
 
 	return nil
 }
+
+// ---------------------------------------------------------------------------------------------------------------------
+
+// BatchBuilder is used for creating the batch operations list. It contains the list of either delete or set tier sub-requests.
+// NOTE: All sub-requests in the batch must be of the same type, either delete or set tier.
+type BatchBuilder struct {
+	endpoint      string
+	authPolicy    policy.Policy
+	subRequests   []*policy.Request
+	operationType *exported.BlobBatchOperationType
+}
+
+func (bb *BatchBuilder) checkOperationType(operationType exported.BlobBatchOperationType) error {
+	if bb.operationType == nil {
+		bb.operationType = &operationType
+		return nil
+	}
+	if *bb.operationType != operationType {
+		return fmt.Errorf("BlobBatch only supports one operation type per batch and is already being used for %s operations", *bb.operationType)
+	}
+	return nil
+}
+
+// BatchDeleteOptions contains the optional parameters for the BatchBuilder.Delete method.
+type BatchDeleteOptions struct {
+	VersionID *string
+	Snapshot  *string
+	*blob.DeleteOptions
+}
+
+func (o *BatchDeleteOptions) format() (*generated.BlobClientDeleteOptions, *generated.LeaseAccessConditions, *generated.ModifiedAccessConditions) {
+	if o == nil {
+		return nil, nil, nil
+	}
+
+	basics := generated.BlobClientDeleteOptions{
+		DeleteSnapshots: o.DeleteSnapshots,
+		DeleteType:      o.BlobDeleteType, // None by default
+		Snapshot:        o.Snapshot,
+		VersionID:       o.VersionID,
+	}
+
+	leaseAccessConditions, modifiedAccessConditions := exported.FormatBlobAccessConditions(o.AccessConditions)
+	return &basics, leaseAccessConditions, modifiedAccessConditions
+}
+
+// BatchSetTierOptions contains the optional parameters for the BatchBuilder.SetTier method.
+type BatchSetTierOptions struct {
+	VersionID *string
+	Snapshot  *string
+	*blob.SetTierOptions
```

Comment:
The convention is to place anonymously embedded types as the first field.

Also, this doesn't need to be pointer-to-type.

Reply:
Done